renegade-crypto/src/ecdsa.rs

```diff
@@ -0,0 +1,204 @@
+//! An implementation of ECDSA over the STARK curve, following the IETF RFC 6979 specification: https://datatracker.ietf.org/doc/html/rfc6979
+
+use ark_ff::PrimeField;
+use itertools::Itertools;
+use mpc_stark::algebra::{scalar::Scalar, stark_curve::StarkPoint};
+use rand::rngs::OsRng;
+use tiny_keccak::{Hasher, Keccak};
+
+/// An ECDSA signature
+pub struct Signature {
+    /// The `r` value of the signature
+    pub r: Scalar,
+    /// The `s` value of the signature
+    pub s: Scalar,
+}
+
+/// Computes a Keccak256 hash over a sequence of bytes, returning a 32-byte hash output
+fn keccak256(bytes: &[u8]) -> [u8; 32] {
+    let mut hash_bytes = [0_u8; 32];
+    let mut hasher = Keccak::v256();
+    hasher.update(bytes);
+    hasher.finalize(&mut hash_bytes);
+    hash_bytes
+}
+
+/// Computes a Keccak256 hash over a message that has been serialized into a sequence of bytes,
+/// and returns the result as a big-endian integer reduced modulo the scalar field order
+pub fn bytes_message_hash_to_scalar(message: &[u8]) -> Scalar {
+    let hash_bytes = keccak256(message);
+    Scalar::from_be_bytes_mod_order(&hash_bytes)
+}
+
+/// Computes a Keccak256 hash over a message that has been serialized into a sequence of scalars
+/// in a manner that is consistent with the Cairo implementation of Keccak256,
+/// and returns the result as a big-endian integer reduced modulo the scalar field order
+pub fn scalar_message_hash_to_scalar(message: &[Scalar]) -> Scalar {
```

Comment:
Referring to the input/output types here convoludes the intent of the function; I'd prefer just `compute_message_hash` and `compute_bytes_hash` above. The I/O types are already part of the signature